modules/RTC/RTCUtils.js

```diff
@@ -749,7 +763,49 @@ class RTCUtils extends Listenable {
         initRawEnumerateDevicesWithCallback();
 
         return new Promise((resolve, reject) => {
-            if (RTCBrowserType.isFirefox()) {
+            if (RTCBrowserType.usesNewGumFlow()) {
+                this.RTCPeerConnectionType = window.RTCPeerConnection;
+
+                this.getUserMedia
```

Comment:
i don't remember why we did this, since i think adapter will do either the callback or the promise (based on how many args were passed), so we should double check

Reply:
It looks to me like one reason it's done this way is to avoid touching screen obtainer, which takes in this.getUserMedia and uses the callback style.

Reply:
Plus I do not see in chrome/getusermedia.js it looking at the arguments to determine which type of gum to use. It looks to me whatever is on navigator.mediaDevices expects constraints to be passed in and a promise to be returned.